### PR TITLE
Add Jaeger tchannel receiver to a new jaeger legacy receiver

### PR DESCRIPTION
**Pull request by objectiser**
**Description:**
Move Jaeger tchannel receiver to contrib repo as step towards removal.

**Link to tracking Issue:**
Related to https://github.com/open-telemetry/opentelemetry-collector/pull/636

**Testing:**
Used existing tchannel related tests from the core repo.

**Documentation:**
No documentation - doesn't seem to be a consistent approach for documenting the contrib receivers?

### CODE REVIEW

File: receiver/jaegerlegacyreceiver/trace_receiver_test.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+					},
+				},
+			},
+			SourceFormat: "jaeger",
+		},
+	}
+}
+
+// TimeToTimestamp converts a time.Time to a timestamp.Timestamp pointer.
+func TimeToTimestamp(t time.Time) *timestamp.Timestamp {
```
Comment (objectiser) [inline receiver/jaegerlegacyreceiver/trace_receiver_test.go]: Note, this was copied from https://github.com/open-telemetry/opentelemetry-collector/blob/master/internal/internal.go#L26-L35 as it didn't like trying to import the internal package.

File: receiver/jaegerlegacyreceiver/factory_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jaegerlegacyreceiver
+
+import (
+	"context"
+	//"fmt"
```
Comment (pjanotti) [inline receiver/jaegerlegacyreceiver/factory_test.go]: nit: delete

File: receiver/jaegerlegacyreceiver/factory.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
+	"github.com/open-telemetry/opentelemetry-collector/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/receiver"
+	"github.com/spf13/viper"
+	"go.uber.org/zap"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "jaegerlegacy"
```
Comment (pjanotti) [inline receiver/jaegerlegacyreceiver/factory.go]: nit: we have been using underscores on this type of config entry
```suggestion
	typeStr = "jaeger_legacy"
```

File: receiver/jaegerlegacyreceiver/go.mod [hunk 4]
```diff
@@ -0,0 +1,17 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/receiver/jaegerlegacyreceiver
+
+go 1.12
```
Comment (pjanotti) [inline receiver/jaegerlegacyreceiver/go.mod]: Any reason to keep this version? Other components are not setting a specific version.

File: receiver/jaegerlegacyreceiver/trace_receiver_test.go (hunk 1 above)
Comment (pjanotti) [reply receiver/jaegerlegacyreceiver/trace_receiver_test.go]: yeah, we need to see some of these common functions and move to the translation package.

File: receiver/collectdreceiver/go.sum [hunk 5]
```diff
@@ -467,6 +467,7 @@ github.com/onsi/gomega v1.5.0 h1:izbySO9zDPmjJ8rDjLvkA2zJHIo+HkYXHnf7eN7SSyo=
 github.com/onsi/gomega v1.5.0/go.mod h1:ex+gbHU/CVuBBDIJjb2X0qEXbFg53c61hWP/1CpauHY=
 github.com/onsi/gomega v1.7.0 h1:XPnZz8VVBHjVsy1vzJmRwIcSwiUO+JFfrv/xGiigmME=
 github.com/onsi/gomega v1.7.0/go.mod h1:ex+gbHU/CVuBBDIJjb2X0qEXbFg53c61hWP/1CpauHY=
+github.com/open-telemetry/opentelemetry-collector v0.2.7 h1:8zWYD+iIkvvV3B0rElgUUmpgujqZyIDQKLOw3lpotzI=
```
Comment (pjanotti) [inline receiver/collectdreceiver/go.sum]: interesting that this showed up here, perhaps something triggered by your IDE?

**Review by pjanotti — COMMENTED** [review]
LGTM @objectiser just a few nits & Qs

File: receiver/jaegerlegacyreceiver/go.mod (hunk 4 above)
Comment (objectiser) [reply receiver/jaegerlegacyreceiver/go.mod]: Done - although the other receivers all have versions defined (some 1.12, others 1.13).

File: receiver/collectdreceiver/go.sum (hunk 5 above)
Comment (objectiser) [reply receiver/collectdreceiver/go.sum]: These were added when doing the build with make.

**Review by pjanotti — APPROVED** [review]
LGTM

**Review by owais — APPROVED** [review]

**Comment by objectiser** [comment]
@pjanotti @owais I think the testbed failure is unrelated - could you kick off CI tests again?

**Pull request merged**